internal/provider/resource_string.go

```diff
@@ -382,143 +388,149 @@ func stringSchemaV3() tfsdk.Schema {
 				},
 			},
 
-			"special": {
+			"special": schema.BoolAttribute{
 				Description: "Include special characters in the result. These are `!@#$%&*()-_=+[]{}<>:?`. Default value is `true`.",
-				Type:        types.BoolType,
 				Optional:    true,
 				Computed:    true,
-				PlanModifiers: []tfsdk.AttributePlanModifier{
-					planmodifiers.DefaultValue(types.BoolValue(true)),
-					planmodifiers.RequiresReplace(),
+				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifiers.DefaultValue(
+						types.BoolValue(true),
+					),
+					boolplanmodifiers.RequiresReplace(),
```

Comment:
Similarly here and below regarding the usage of the framework-based plan modifiers